docs/framework/guides/deep-dive/localization.md

```diff
@@ -0,0 +1,240 @@
+
+When using the `t()` function, you can create your own *localizable messages* or reuse *messages* created in CKEditor 5 packages that your project depends on. In case of reusing *messages*, you will not need to worry about translating them as all work will be done by the CKEditor 5 team and [Transifex translators](https://www.transifex.com/ckeditor/ckeditor5/) anyway. Obviously, your help in translating will still be appreciated!
+
+For simple *localizable messages*, use the string form for simplicity:
+
+```js
+const emojiName = 'cat';
+
+// Assuming that the English language was picked:
+t( 'insert emoji' ); // "insert emoji"
+t( 'insert %0 emoji', emojiName ); // "insert cat emoji"
+t( 'insert %0 emoji', [ emojiName ] ); // "insert cat emoji"
+```
+
+For more advanced scenarios, use plain object forms:
+
+```js
+const quantity = 3;
+
+// Assuming that the English language was picked:
+t( { string: '%0 emoji', id: 'INSERT_EMOJI' }, 'insert' ); // "insert emoji"
+t( { string: '%0 emoji', plural: '%0 emojis', id: 'EMOJI' }, quantity ); // "3 emojis"
+t( { string: '%1 %0 emoji', plural: '%1 %0 emojis', id: 'INSERT_EMOJIS' }, [ quantity, 'Insert' ] ); // "Insert 3 emojis"
+```
+
+<info-box warning>
+	Note that the `t()` function should not be used as a method of the `Locale` class because during the translation process a static code analyzer catches *localizable messages* only from function forms. For the same reason its first argument accepts only a string literal or an object literal. `Locale#t()` method calls and incompatible expressions passed to the `t()` calls will produce build-time warnings.
+</info-box>
+
+### Example: Localizing the plugin UI
+
+This example shows how to create a localizable user interface of a plugin.
+
+```js
+// ...
+editor.ui.componentFactory.add( 'smilingFaceEmoji', locale => {
+	const command = editor.commands.get( 'insertSmilingFaceEmoji' );
+	const buttonView = new ButtonView( locale );
+
+	// The translation function.
+	const { t } = editor.locale;
+
+	// The localized label.
+	const label = t( 'insert smiling face emoji' );
+
+	buttonView.set( {
+		label,
+		icon: emojiIcon,
+		tooltip: true
+	} );
+
+	buttonView.on( 'execute', () => {
+		editor.execute( 'insertSmilingFaceEmoji' );
+		editor.editing.view.focus();
+	} );
+} );
+// ...
+```
+
+<info-box warning>
+	Note that this code sample lacks a few parts. See {@link framework/guides/creating-simple-plugin how to create a complete plugin} to have a better understanding about creating CKEditor 5 plugins.
+</info-box>
+
+### Example: Localizing `aria` attributes
+
+For better accessibility support, do not forget about making `aria` attributes localizable:
+
+```js
+editingView.change( writer => {
+	const editingView = this._editingView;
+	const t = this.t;
```

Comment:
What is `this` here?

In general, I don't understand this example, TBH. And there is a problem at the beginning (you have `editingView` in the first line and then you declare a const with that name).

I'd either make it example more elaborate or skip it altogether (it doesn't add much when it comes to translations system).